tokio/docs/runtime-stats.md

```diff
@@ -0,0 +1,137 @@
+# Tokio Runtime Stats
+
+## Summary
+
+This RFC proposes a new way to gather understanding from the Tokio runtime. Currently, the runtime does not expose any methods to understand what is happening under the hood. This provides a rough experience when deploying Tokio based applications into production where you would like to understand what is happening to your code. Via this RFC, we will propose a few methods to collect this data at different levels. Beyond what is proposed as implemenation in this RFC, we will also discuss other methods to gather the information a user might need to be successful with Tokio in a production environment.
+
+There are two main types of stats that a runtime can expose. A per-runtime stats (eg `executor_load`, `fds_registered`) that are collected indepdently of the tasks running on the runtime. A per-task stats (eg `poll_duration`, `amount_polls`) that are collected and aggregated at the task level. This RFC will propose an implemenation for implementing per-runtime stats but will also mention methods to capture per-task stats.
+
+A small note, the term `stats` is used instead of `metrics` because we are only concerned with exposing raw data rather than methods of aggregating and emiting that data.
```

Comment:
This was odd to me. "stats" is short for "statistics", which are just as much aggregated as "metrics" are. Would "performance counters" be better"? Or "performance events"? Or maybe "observations" or simply "data"?

Reply:
+1 for "metrics"; I'm not sure the distinction here matters that much.  People will want the metrics, try to figure out how it works, see they need an extra crate, and be on their way.  Calling it "stats" doesn't imply that, we'll have to spell it out in the documentation.  So just go with the more common term of "metrics," IMO.